File: app/src/main/java/br/com/star_wars_wiki/database/dao/FavoriteDAO.java
package br.com.star_wars_wiki.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import br.com.star_wars_wiki.entity.Favorite;

@Dao
public interface FavoriteDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFavorite(Favorite favorite);

    @Delete
    void removeFavorite(Favorite favorite);

    @Query("SELECT * FROM Favorite")
    LiveData<List<Favorite>> getAllFavorites();
}

File: app/src/main/java/com/matheussilas97/starwarsapp/database/dao/FavoriteDAO.kt
package com.matheussilas97.starwarsapp.database.dao

import androidx.room.*
import com.matheussilas97.starwarsapp.database.model.FavoriteModel

@Dao
interface FavoriteDAO {

    @Query("SELECT * FROM favorites")
    fun getAllFavorites(): List<FavoriteModel>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFavorite(item: FavoriteModel): Long

    @Delete
    fun deleteFavorite(item: FavoriteModel)

    @Query("SELECT * FROM favorites WHERE id = :id")
    fun load(id: String): FavoriteModel?

    @Query("SELECT * FROM favorites WHERE url = :url")
    fun isFavorite(url: String): Boolean
}